### PR TITLE
feat: add terraform to bundle deploy

### PR DESCRIPTION
**Pull request by marcoppenheimer**
##### `feat: add ability to use terraform`
- Updates `.gitignore` to not track state files
- Updates `README.md` to include install + deploy instructions
- Adds `terraform/*` for `dev/prod` deployments:
    - Symlinks to root directory as per TF best practices
    - Uesr configuration managed in `<env>.auto.tfvars`
    - Default configuration managed in `vars.tf`
    - Specifies machine constraints, which currently the Juju bundle equivalent **does not** have
    - Is unable to set storage constraints (I think), which currently the Juju bundle equivalent **does** have
    - Deploys a `data-integrator` by default
- Added a `terraform fmt` to `tox -e fmt`. Nicer style

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,31 +7,45 @@ The steps outlined below are based on the assumption that you are deploying the
...
+# (required) installing lxd and juju
 sudo snap install lxd --classic
 sudo snap install juju --classic
+
+# (optional) installing terraform
+sudo apt install gpg
+wget -O- https://apt.releases.hashicorp.com/gpg | sudo gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/hashicorp.list
+sudo apt update
+sudo apt install terraform
```
Comment (marcoppenheimer) [inline README.md:19]: **question:** Am not an Ubuntu user, tested this in a VM but if anyone can check this that would be great.

Comment (jnsgruk) [reply README.md:19]: I'd replace this with `sudo snap install terraform --classic` :)

File: terraform/applications.tf [hunk 2]
```diff
@@ -0,0 +1,111 @@
+resource "juju_application" "kafka" {
+  depends_on = [juju_machine.kafka]
+  model      = juju_model.kafka.name
```
Comment (jnsgruk) [inline terraform/applications.tf:3]: I  could be wrong, but could this not be simplified to `juju_model.kafka`?

File: terraform/applications.tf [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+  units = var.kafka.units
+
+  charm {
+    name    = "kafka"
+    channel = var.kafka.channel
+    series  = var.kafka.series
+  }
+
+  placement = join(",", [for machine in juju_machine.kafka : split(":", machine.id)[1]])
```
Comment (jnsgruk) [inline terraform/applications.tf:13]: While it's relatively clear to someone who knows HCL, I'd probably add a comment to lines like this to explain the transformation.

**Review by jnsgruk — APPROVED** [review]
LGTM, nice work :+1:

**Comment by deusebio** [comment]
Looks very good! Just before merging, could we maybe just put this deployment into a small integration tests where we just do 

```
terraform apply
```

and make sure that everything comes up? A bit like what we do for bundles as well?

**Comment by marcoppenheimer** [comment]
> Looks very good! Just before merging, could we maybe just put this deployment into a small integration tests where we just do
> 
> ```
> terraform apply
> ```
> 
> and make sure that everything comes up? A bit like what we do for bundles as well?

@deusebio Done

**Review by deusebio — APPROVED** [review]
LGTM! Great to have this test as well!

**Pull request merged**